src/Unleash.php

```diff
@@ -13,43 +12,38 @@
 
 class Unleash
 {
-    protected $client;
+    const DEFAULT_CACHE_TTL = 15;
 
+    protected $client;
     protected $cache;
-
     protected $config;
-
     protected $request;
-
-    protected $features = [];
+    protected $features;
 
     public function __construct(ClientInterface $client, Cache $cache, Config $config, Request $request)
     {
         $this->client = $client;
         $this->cache = $cache;
         $this->config = $config;
         $this->request = $request;
-
-        if (!$this->config->get('unleash.isEnabled')) {
-            return;
-        }
-
-        if ($this->config->get('unleash.cache.isEnabled')) {
-            $this->features = $this->cache->remember(
-                'unleash',
-                $this->config->get('unleash.cache.ttl'),
-                function () {
-                    return $this->fetchFeatures();
-                }
-            );
-        } else {
-            $this->features = $this->fetchFeatures();
-        }
     }
 
     public function getFeatures(): array
     {
-        return $this->features;
+        try {
+            $features = $this->getCachedFeatures();
+
+            // Always store the failover cache, in case it is turned on during failure scenarios.
+            $this->cache->forever('unleash.features.failover', $features);
+
+            return $features;
+        } catch (RequestException | \RuntimeException $e) {
```

Comment:
Bit concerned about catching `RuntimeException`s here, IMO all I would only want to catch request failures to Unleash. I feel like this could make things hard to debug?

Reply:
Resolved to only catch Guzzle or JSON exceptions (the latter of which I was throwing RuntimeExceptions for)